### ISSUE TITLE
Feature: Prioritized Reset Sampling for Coverage-Based Training

### ISSUE THREAD

**Issue by BY571** [TorchTrade/torchtrade#45]
### Description

Implement prioritized reset sampling to encourage exploration of under-visited regions of the dataset during training. This feature builds on the coverage tracking metric from issue #41 and enables curriculum-style learning.

### Motivation

During RL training with random episode resets, certain starting positions may be visited more frequently than others due to pure randomness. This can lead to:
- Overfitting on frequently-sampled market conditions
- Under-exploration of rare but important scenarios
- Uneven learning across the dataset

By tracking which starting positions have been visited and prioritizing under-explored regions, we can:
- Ensure more comprehensive coverage of training data
- Improve generalization to diverse market conditions
- Enable curriculum learning strategies

#### 1. Coverage Tracking Foundation

First, implement coverage tracking in `MarketDataObservationSampler` (from issue #41):

```python
class MarketDataObservationSampler:
    def __init__(self, ...):
        # Existing code...

        # Coverage tracking
        self.coverage_counts = np.zeros(len(self.exec_times), dtype=np.int32)
        self.total_resets = 0

    def reset(self, random_start: bool = False) -> int:
        # Track which start_idx was selected
        # ... existing reset logic ...

        self.coverage_counts[start_idx] += 1
        self.total_resets += 1
        return self._end_idx - self._sequential_idx
```

#### 2. Prioritized Sampling Strategy

Add prioritized reset mode that samples under-visited positions with higher probability:

```python
class MarketDataObservationSampler:
    def reset(
        self,
        random_start: bool = False,
        prioritized: bool = False,
        epsilon: float = 0.1
    ) -> int:
        """
        Reset the sampler.

        Args:
            random_start: If True, start from random position
            prioritized: If True, prioritize under-visited positions
            epsilon: Probability of uniform random sampling (exploration)
        """
        total_len = len(self._exec_times_arr)

        if random_start:
            if self.max_traj_length is None:
                max_start_index = total_len - 1
            else:
                max_start_index = max(0, total_len - self.max_traj_length)

            if prioritized and self.total_resets > 0:
                # Epsilon-greedy: sometimes uniform, sometimes prioritized
                if self.np_rng.random() < epsilon:
                    # Uniform random (exploration)
                    start_idx = int(self.np_rng.integers(0, max_start_index + 1))
                else:
                    # Prioritized sampling (exploitation of under-explored regions)
                    # Compute inverse frequency weights
                    weights = 1.0 / (self.coverage_counts[:max_start_index + 1] + 1)
                    probabilities = weights / weights.sum()
                    start_idx = self.np_rng.choice(max_start_index + 1, p=probabilities)
            else:
                # Standard uniform random
                start_idx = int(self.np_rng.integers(0, max_start_index + 1))

            self._sequential_idx = start_idx
            if self.max_traj_length is None:
                self._end_idx = total_len
            else:
                self._end_idx = min(start_idx + self.max_traj_length, total_len)
        else:
            # Sequential start
            start_idx = 0
            self._sequential_idx = 0
            # ... existing logic ...

        # Track coverage
        self.coverage_counts[start_idx] += 1
        self.total_resets += 1

        return self._end_idx - self._sequential_idx
```

#### 3. Environment Configuration

Add prioritized sampling config to environment configs:

```python
@dataclass
class SeqLongOnlyEnvConfig:
    # ... existing fields ...
    random_start: bool = True
    prioritized_sampling: bool = False
    prioritized_epsilon: float = 0.1
```

Update environment reset:

```python
class SeqLongOnlyEnv(EnvBase):
    def _reset(self, tensordict: TensorDictBase, **kwargs) -> TensorDictBase:
        # ... existing reset logic ...

        max_episode_steps = self.sampler.reset(
            random_start=self.random_start,
            prioritized=self.config.prioritized_sampling,
            epsilon=self.config.prioritized_epsilon
        )

        # ... rest of reset ...
```

#### 4. Curriculum Learning Strategies

Support multiple prioritization strategies:

```python
class PrioritizationStrategy(Enum):
    UNIFORM = "uniform"              # Standard random
    INVERSE_FREQUENCY = "inverse"    # Prioritize least-visited
    VARIANCE_WEIGHTED = "variance"   # Prioritize high-variance returns
    RECENT_ERROR = "error"           # Prioritize recent high TD-error
```

### Integration with Training

Enable via Hydra config:

```yaml
env:
  random_start: true
  prioritized_sampling: true
  prioritized_epsilon: 0.1
  prioritization_strategy: "inverse"
```

### Metrics to Track

During training, log:
- Coverage entropy (measure of distribution uniformity)
- Prioritization hit rate (% of resets using prioritized vs random)
- Average visits per position over time

### Benefits

1. **Better Generalization**: More even coverage reduces overfitting
2. **Curriculum Learning**: Can start uniform, gradually increase prioritization
3. **Active Learning**: Focus on regions where agent is uncertain
4. **Data Efficiency**: Learn from under-explored regions faster

### Implementation Checklist

- [ ] Add coverage tracking to `MarketDataObservationSampler`
- [ ] Implement prioritized reset sampling with epsilon-greedy
- [ ] Add configuration parameters to environment configs
- [ ] Support multiple prioritization strategies
- [ ] Add logging of prioritization metrics
- [ ] Write tests for prioritized sampling
- [ ] Update documentation with usage examples
- [ ] Add example training script demonstrating curriculum learning

### Related Issues

- #41 - Coverage Metric (prerequisite)

### References

- [Experience Replay with Prioritized Sampling](https://arxiv.org/abs/1511.05952)
- [Reverse Curriculum Generation for RL](https://arxiv.org/abs/1707.05300)

---

**Note**: This feature should be implemented after coverage tracking (#41) is complete, as it depends on the coverage counts infrastructure.
